### PR TITLE
AdminBookmController.reattributeKeeps

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepSourceCommander.scala [hunk 1]
```diff
@@ -72,12 +72,12 @@ class KeepSourceCommanderImpl @Inject() (
...
   }
 
-  def reattributeKeeps(author: Author, userId: Id[User]): Set[Id[Keep]] = {
+  def reattributeKeeps(author: Author, userId: Id[User], overwriteExistingOwner: Boolean = false): Set[Id[Keep]] = {
     val keepIds = db.readOnlyMaster { implicit session => sourceAttributionRepo.getKeepIdsByAuthor(author) }
     keepIds.grouped(100).flatMap { batchedKeepIds =>
       db.readWrite { implicit s =>
         keepRepo.getByIds(batchedKeepIds).values.collect {
-          case keep if keep.userId.isEmpty => keepCommander.setKeepOwner(keep, userId).id.get
+          case keep if keep.userId.isEmpty || overwriteExistingOwner => keepCommander.setKeepOwner(keep, userId).id.get
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepSourceCommander.scala:80]: :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala [hunk 2]
```diff
@@ -328,4 +332,25 @@ class AdminBookmarksController @Inject() (
 
     Ok(keeps.length.toString)
   }
+
+  def reattributeKeeps(author: String, overwriteExistingOwner: Boolean, doIt: Boolean) = AdminUserAction { implicit request =>
+    Try(Author.fromIndexableString(author)).toOption match {
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala:337]: I considered giving `fromIndexableString` a return type of `Option[Author]` but decided against it for simplicity. Thoughts?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepSourceCommander.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepSourceCommander.scala:80]: Look in `KeepCommander.setKeepOwner`, pretty sure there's a `require` there that's going to make you sad.

**Comment by ryanpbrewster** [comment]
lgtm, be sure to remove the `require` in keep commander

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala:337]: Probably fine for now

**Pull request merged**